mlir-tensorrt/compiler/lib/Dialect/TensorRTRuntime/Transforms/BufferizableOpInterfaceImpl.cpp

```diff
@@ -153,18 +160,144 @@ struct EnqueueOpInterface
     rewriter.create<EnqueueOp>(
         op->getLoc(), newInputBuffers[0], newInputBuffers[1],
         ValueRange(newInputBuffers).drop_front(2), newOutputBuffers,
-        callOp.getHostTensorArgsAttr());
+        enqueueOp.getHostTensorArgsAttr());
     replaceOpWithBufferizedValues(rewriter, op, newOutputBuffers);
     return success();
   }
 };
 
+
+struct EnqueueAllocOpInterface
+    : public bufferization::BufferizableOpInterface::ExternalModel<
+          EnqueueAllocOpInterface, EnqueueAllocOp> {
+  bool bufferizesToMemoryRead(Operation *op, OpOperand &opOperand,
+                              const bufferization::AnalysisState &state) const {
+    auto enqueueAllocOp = cast<EnqueueAllocOp>(op);
+    OperandRange inputs = enqueueAllocOp.getInputs();
+    return std::find(inputs.begin(), inputs.end(), opOperand.get()) !=
+           inputs.end();
+  }
+
+  bool bufferizesToMemoryWrite(Operation *op, OpOperand &opOperand,
+                               const bufferization::AnalysisState &state) const {
+    return false; // This op doesn't write to its inputs
+  }
+  
+
+  bufferization::AliasingValueList getAliasingValues(Operation *op, OpOperand &opOperand,
+                                      const bufferization::AnalysisState &state) const {
+    return {};
+  }
+
+  bool mustBufferizeInPlace(Operation *op, OpOperand &opOperand,
+                            const bufferization::AnalysisState &state) const {
+    // EnqueueAllocOp creates new outputs, doesn't modify inputs in-place
+    return false;
+  }
+
+  SmallVector<OpResult> getAliasingOpResult(Operation *op, OpOperand &opOperand,
+                                            const bufferization::AnalysisState &state) const {
+    return {}; // This op doesn't alias its inputs to its outputs
+  }
+
+  bool bufferizesToElementwiseAccess(Operation *op,
+                                     const bufferization::AnalysisState &state,
+                                     ArrayRef<OpOperand *> opOperands) const {
+    return true;
+  }
+
+  LogicalResult bufferize(Operation *op, RewriterBase &rewriter,
+                          const BufferizationOptions &options) const {
+    auto enqueueAllocOp = cast<EnqueueAllocOp>(op);
+    MLIRContext *ctx = op->getContext();
+    Location loc = op->getLoc();
+    rewriter.setInsertionPoint(enqueueAllocOp);
+
+    // Handle inputs
+    SmallVector<Value> newInputBuffers;
+    newInputBuffers.reserve(enqueueAllocOp.getInputs().size());
+    for (auto [idx, input] : llvm::enumerate(enqueueAllocOp.getInputs())) {
+      if (isScalar(&input)) {
```

Comment:
This operation cannot have scalar inputs by construction. Values in `getInputs()` can only be tensors or memrefs. This can just be

```
assert(isa<RankedTensorType>(input.getType()) && "expected tensor inputs");
```